### PR TITLE
META-4203 Add keycloak Admin client

### PR DESCRIPTION
**Pull request by nikhilbonte21**

### CODE REVIEW

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            try {
+                initConf();
+            } catch (IOException e) {
+                LOG.error("Failed to fetch Keycloak conf {}", e.getMessage());
+            } catch (JSONException e) {
+                LOG.error("Failed to parsing Keycloak conf {}", e.getMessage());
+            } catch (Exception e) {
+                LOG.error("Failed to connect to Keycloak {}", e.getMessage());
+            }
+            init();
```
Comment (mehtaanshul) [inline client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java:75]: Should we still call the init method, if there's an error in initConf?

Comment (nikhilbonte21) [reply client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java:78]: Fixed, throwing exception.

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            LOG.info("Initialized Keycloak client..");
+        }
+
+        if (keycloak.isClosed()) {
+            LOG.info("Re-initializing keycloak client");
+            init();
+        }
+
+        return keycloakClient;
```
Comment (n5nk) [inline client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java:87]: Check if `@PreDestroy` gets called here, if yes, close `keycloakClient`

File: client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        } while (CollectionUtils.isNotEmpty(ret) && ret.size() % size == 0);
+
+        return ret;
+    }
+
+    public List<RoleRepresentation> getAllRoles() {
+        int start = 0;
+        int size = 100;
+
+        List<RoleRepresentation> ret = new ArrayList<>();
```
Comment (n5nk) [inline client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java:175]: nit: Always init `ArrayList` with size 0 => `new ArrayList<>(0);` in all places.

**Review by n5nk — COMMENTED** [review]
minor review comments.

**Pull request merged**